File: app/src/main/java/booboo/thelocalnick/CreateTour/PhotoViewHolder.java
package booboo.thelocalnick.CreateTour;

import android.databinding.BindingAdapter;
import android.databinding.DataBindingUtil;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.ImageView;

import booboo.thelocalnick.data.Photo;
import booboo.thelocalnick.databinding.PhotoCardViewBinding;

class PhotoViewHolder extends RecyclerView.ViewHolder {
    private PhotoCardViewBinding photoCardViewBinding;

    public PhotoViewHolder(View view) {
        super(view);
        photoCardViewBinding = DataBindingUtil.bind(view);
    }

    public void bind(Photo photo) {
        photoCardViewBinding.setPhoto(photo);
    }

    @BindingAdapter({"android:src"})
    public static void setImageViewResource(ImageView imageView, int resource) {
        imageView.setImageResource(resource);
    }
}

File: app/src/main/java/booboo/thelocalnick/CreateTour/PhotoViewHolder.kt
package booboo.thelocalnick.CreateTour

import android.databinding.DataBindingUtil
import android.support.v7.widget.RecyclerView
import android.view.View
import booboo.thelocalnick.data.Spot

import booboo.thelocalnick.databinding.PhotoCardViewBinding

class PhotoViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    private val photoCardViewBinding: PhotoCardViewBinding = DataBindingUtil.bind<PhotoCardViewBinding>(itemView)

    fun bind(spot: Spot) {
        photoCardViewBinding.spot = spot
    }
}
